### PR TITLE
Add continuous integration support via GitHub Actions

### PR DESCRIPTION
**Pull request by hectcastro**
Exercise the project initialization process via continuous integration on every PR and merges to `master`. In addition:

- Add support for keeping GitHub Actions up-to-date with Dependabot
- Add support for automatically merging Dependabot PRs that pass CI
- Add support for marking inactive issues as stale, and eventually closing them
- Update `sbt-extras` script
- Fix the Typesafe repository resolver

---

**Testing**

Review the [build output](https://github.com/azavea/azavea.g8/pull/22/checks) associated with this PR.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
```
Comment (hectcastro) [inline .github/dependabot.yml:3]: This helps keep GitHub Actions up-to-date via Dependabot PRs.

File: .github/workflows/continuous_integration.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      fail-fast: false
+      matrix:
+        jdk-version: [adopt@1.11]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Scala ${{ matrix.jdk-version }}
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: ${{ matrix.jdk-version }}
```
Comment (hectcastro) [inline .github/workflows/continuous_integration.yml:23]: It appears that Jabba is being used under the covers by this Action. If we want to test against additional JDK versions, we can add them to the `jdk-version` matrix list above.

File: .github/workflows/continuous_integration.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+          key: sbt-${{ hashFiles('**/build.sbt') }}
+          restore-keys: sbt-
+
+      - name: Create new project from template
+        run: |
+          sbt new file://${GITHUB_WORKSPACE} \
```
Comment (hectcastro) [inline .github/workflows/continuous_integration.yml:36]: `GITHUB_WORKSPACE` here corresponds to the project directory checked out by `actions/checkout`.

File: .github/workflows/merge.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+name: Automatically merge Dependabot PRs
+
+on:
+  check_suite:
+    types:
+      - completed
+
+jobs:
```
Comment (hectcastro) [inline .github/workflows/merge.yml:8]: One of these is responsible for `dependabot.yml` initiated PRs and the other is for GitHub Security PRs.

File: build.sbt [hunk 5]
```diff
@@ -9,5 +9,5 @@ lazy val root = (project in file("."))
       val _ = (g8Test in Test).toTask("").value
     },
     scriptedLaunchOpts ++= List("-Xms1024m", "-Xmx1024m", "-XX:ReservedCodeCacheSize=128m", "-Xss2m", "-Dfile.encoding=UTF-8"),
-    resolvers += Resolver.url("typesafe", url("http://repo.typesafe.com/typesafe/ivy-releases/"))(Resolver.ivyStylePatterns)
+    resolvers += Resolver.url("typesafe", url("https://repo.typesafe.com/typesafe/ivy-releases/"))(Resolver.ivyStylePatterns)
```
Comment (hectcastro) [inline build.sbt:12]: This was leading to connection refused errors during dependency resolution.

File: sbt [hunk 6]
```diff
@@ -34,8 +34,8 @@
 
 set -o pipefail
 
-declare -r sbt_release_version="1.3.13"
```
Comment (hectcastro) [inline sbt:37]: All of the changes here are from pasting in the most up-to-date version of `sbt-extras`.

**Review by jisantuc — COMMENTED** [review]
Would it also make sense to copy this configuration into the project to create (under `src/main/g8`)? If people want to use a different CI, they can remove it, but if not we can include a note in the template setup about setting the appropriate secret in github

**Comment by hectcastro** [comment]
> Would it also make sense to copy this configuration into the project to create (under `src/main/g8`)? If people want to use a different CI, they can remove it, but if not we can include a note in the template setup about setting the appropriate secret in github

Before marking the PR ready for review, I went through the issue list and almost linked https://github.com/azavea/azavea.g8/issues/1 to this PR as resolved. I ultimately didn't because it seemed like two different concerns: CI for the project template vs. CI for instances of the project template.

I'm personally interested in resolving https://github.com/azavea/azavea.g8/issues/1, but I think doing so in a separate PR makes sense since I think the testing cycle for it will be more involved.

BTW, are you OK with me assigning you to review this PR?

**Comment by jisantuc** [comment]
That makes sense to me and I'm happy to review this

**Review by jisantuc — APPROVED** [review]

**Pull request merged**